modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/ExtensionDefinitionParser.java

```diff
@@ -743,71 +743,90 @@ private boolean isExpressionFunction(MetadataType metadataType)
         }
     }
 
-    private ValueResolver parseDateTime(Object value, DateTimeType dataType, Object defaultValue)
+
+    private static ValueResolver doParseDate(Object value, Class<?> type)
     {
-        Class<?> type = getType(dataType);
-        if (isExpression(value, parser))
-        {
-            return new TypeSafeExpressionValueResolver((String) value, type);
-        }
 
-        if (type.equals(LocalDateTime.class))
+        if (value instanceof String)
         {
-            Date date = doParseDate(value, DATE_FORMAT, defaultValue);
-            return new StaticValueResolver(LocalDateTime.ofInstant(date.toInstant(), ZoneId.systemDefault()));
+            Object constructedValue = null;
+            DateTime dateTime = getParsedDateTime((String) value);
+
+            if (type.equals(LocalDate.class))
+            {
+                constructedValue = LocalDate.of(dateTime.getYear(), dateTime.getMonthOfYear(), dateTime.getDayOfMonth());
```

Comment:
LocalDate jdoc says:

{@code LocalDate} is an immutable date-time object that represents a date, often viewed as year-month-day. This class does not store or represent a time or time-zone.